src/libfmt_macros/lib.rs

```diff
@@ -155,44 +156,38 @@ pub struct Parser<'a> {
     /// `Some(raw count)` when the string is "raw", used to position spans correctly
     style: Option<usize>,
     /// Start and end byte offset of every successfully parsed argument
-    pub arg_places: Vec<(SpanIndex, SpanIndex)>,
+    pub arg_places: Vec<Span>,
     /// Characters that need to be shifted
     skips: Vec<usize>,
-    /// Span offset of the last opening brace seen, used for error reporting
-    last_opening_brace_pos: Option<SpanIndex>,
+    /// Span of the last opening brace seen, used for error reporting
+    last_opening_brace: Option<Span>,
     /// Wether the source string is comes from `println!` as opposed to `format!` or `print!`
     append_newline: bool,
 }
 
-#[derive(Clone, Copy, Debug)]
-pub struct SpanIndex(pub usize);
-
-impl SpanIndex {
-    pub fn unwrap(self) -> usize {
-        self.0
-    }
-}
-
 impl<'a> Iterator for Parser<'a> {
     type Item = Piece<'a>;
 
     fn next(&mut self) -> Option<Piece<'a>> {
         if let Some(&(pos, c)) = self.cur.peek() {
             match c {
                 '{' => {
-                    let curr_last_brace = self.last_opening_brace_pos;
-                    self.last_opening_brace_pos = Some(self.to_span_index(pos));
+                    let curr_last_brace = self.last_opening_brace;
+                    let byte_pos = self.to_span_index(pos);
+                    self.last_opening_brace =
+                        Some(Span::new(byte_pos, byte_pos, SyntaxContext::empty()));
```

Comment:
I think this needs to be `Span::new(byte_pos, byte_pos + 1, SyntaxContext::empty())` if we want the `Span` to point at the `}` and not the space between whatever comes right before the brace and the brace itself.

Reply:
I'll wait to do anything here pending resolution of whether we want to be using Spans at all, though I think this is strictly "the same" as what we had before in any case so it's at least consistent if, well, wrong. :)

Reply:
Hurray for uncovering subtle bugs that would otherwise had gone undetected.